app/service/rest.py

```diff
@@ -303,6 +308,22 @@ def update_service(service_id):
     if service_going_live:
         _warn_services_users_about_going_live(service_id, current_data)
 
+    if current_app.config["FF_SALESFORCE_CONTACT"] and (service_going_live or service_requested_go_live):
+        try:
+            # Two scenarios, if there is a user that has requested to go live, we will use that user
+            # to create a Contact/Engagment pair between Notify and Salesforce.
+            # If by any chance there is no tracked request to a user, Notify will try to identify the user
+            # that created the service and then create a Contact/Engagment relationship.
+            if service.go_live_user_id:
+                user = get_user_by_id(service.go_live_user_id)
+            else:
+                user = dao_fetch_service_creator(service.id)
+
+            stage_name = ENGAGEMENT_STAGE_LIVE if service_going_live else ENGAGEMENT_STAGE_ACTIVATION
```

Comment:
I may be missing something, but I don't think this code will run when a "go live" request is made. "Go live" requests hit a different endpoint: https://github.com/cds-snc/notification-api/blob/main/app/user/rest.py#L438

Reply:
However this code will get executed when a platform admin user makes a service live.

Reply:
In my local testing, the request to "go live" was also hitting that endpoint because of this service update call:
https://github.com/cds-snc/notification-admin/blob/d587ea1169f6813c17a3c6a86c76720fd620a357/app/main/views/service_settings.py#L356